static/js/narrow.js

```diff
@@ -386,6 +392,67 @@ exports.update_selection = function (opts) {
     unread_ops.process_visible();
 };
 
+exports.to_compose_target = function () {
+    if (compose_state.get_message_type() === 'stream' && compose_state.stream_name() !== '') {
+        var stream = compose_state.stream_name();
+        var topic = compose_state.subject();
+        exports.by_stream_topic_name(stream, topic);
+        return true;
+    }
+    if (compose_state.get_message_type() === 'private') {
+        exports.by_recipient_name(compose_state.recipient());
+        return true;
+    }
+    return false;
+};
+
+exports.by_stream_topic_name = function (stream, topic) {
+    var stream_id = stream_data.get_stream_id(stream);
+
+    // If stream is incorrect, we dont do anything.
+    if (stream_id === undefined) {
+        return;
+    }
+
+    var topics = topic_data.get_recent_names(stream_id);
+
+    var opts = {
+        select_first_unread: true,
+        trigger: 'compose_hotkey',
+    };
+    // Narrow to stream only, if topic fails the check below.
+    var raw_operators = [
+        {operator: 'stream', operand: stream},
+    ];
+
+    // If topic is not null and exists in the stream,
+    // We narrow to topic, else stream.
+    if (topic !== '' && _.indexOf(topics, topic) >= 0) {
+        raw_operators.push({operator: 'topic', operand: topic});
+    }
```

Comment:
@rishig @showell do we want this to narrow to the stream if there is no recent topic?  I had sorta imagined we always wanted the stream unless topic was `''`.

Reply:
yeah, I think narrowing to stream makes sense

Reply:
I personally think you should narrow to the topic, even if it doesn't exist, so that you're aware you have an empty topic.  (You might be fishing to see if you have a "clean" topic.).  I don't have a strong opinion, though, and see the opposite argument.

Reply:
I agree with Steve on this and am keeping it to narrow to an empty topic as well.

Reply:
I haven't ever run into a situation where I didn't know I had an empty topic, and in any case, narrowing to stream would still tell you that. I think narrowing to stream here makes the most sense.

Reply:
After testing, I find it comforting to have an empty message list when I expect to be creating a new topic. I'm keeping both versions in separate commits and would encourage you to try the version that goes to the empty topic.